test/v3/createPool/stableSurge/stableSurge.integration.test.ts

```diff
@@ -0,0 +1,179 @@
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+                {
+                    address: scDAI.address,
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+            ],
+            swapFeePercentage: parseEther('0.01'),
+            pauseManager: testAddress,
+            swapFeeManager: testAddress,
+            disableUnbalancedLiquidity: false,
+            enableDonation: false,
+        };
+
+        poolAddress = await doCreatePool({
+            client,
+            testAddress,
+            createPoolInput,
+        });
+    });
+    test('Deployment', async () => {
+        expect(poolAddress).to.not.be.undefined;
+    }, 120_000);
+
+    test('Registration', async () => {
+        const isPoolRegistered = await client.readContract({
+            address: VAULT_V3[chainId],
+            abi: vaultExtensionAbi_V3,
+            functionName: 'isPoolRegistered',
+            args: [poolAddress],
+        });
+        expect(isPoolRegistered).to.be.true;
+    }, 120_000);
+
+    test('Initialization', async () => {
+        await setTokenBalances(
+            client,
+            testAddress,
+            [scDAI.address, scUSD.address],
+            [scDAI.slot!, scUSD.slot!],
+            [parseUnits('100', 18), parseUnits('100', 18)],
+        );
+
+        await approveSpenderOnTokens(
+            client,
+            testAddress,
+            [scDAI.address, scUSD.address],
+            PERMIT2[chainId],
+        );
+
+        const initPoolInput = {
+            amountsIn: [
+                {
+                    address: scUSD.address,
+                    rawAmount: parseUnits('100', scUSD.decimals),
+                    decimals: scUSD.decimals,
+                },
+                {
+                    address: scDAI.address,
+                    rawAmount: parseUnits('100', scDAI.decimals),
+                    decimals: scDAI.decimals,
+                },
+            ],
+            minBptAmountOut: parseEther('10'),
+            chainId,
+        };
+
+        const initPool = new InitPool();
+        const permit2 = await Permit2Helper.signInitPoolApproval({
+            ...initPoolInput,
+            client,
+            owner: testAddress,
+        });
+        const initPoolBuildOutput = initPool.buildCallWithPermit2(
+            initPoolInput,
+            {
+                id: poolAddress,
+                address: poolAddress,
+                type: poolType,
+                protocolVersion: 3,
+                tokens: [
+                    {
+                        ...scUSD,
+                        index: 0,
+                    },
+                    {
+                        ...scDAI,
+                        index: 1,
+                    },
+                ],
+            },
+            permit2,
+        );
```

Comment:
Curious about the decision to avoid using the poolState helper here. Fewer reads to optimize the speed of sluggish test?

https://github.com/balancer/b-sdk/blob/4f07d199b78720816c63743efd6afae67aa7f21f/test/v3/createPool/stable/stable.integration.test.ts#L110-L116

https://github.com/balancer/b-sdk/blob/4f07d199b78720816c63743efd6afae67aa7f21f/test/v3/createPool/stable/stable.integration.test.ts#L156-L160